### PR TITLE
[feature/feedback] Add Help & Support overview, clarify Feedback option

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
- new view providing an overview of help and support options with a brief textual description, adding the forums to the mix
- for non-branded apps Settings make this view available as "Help & Contact", replacing "Documentation", "Help" and "Feedback"

## Related Issue
https://github.com/owncloud/enterprise/issues/6594

## Screenshots (if appropriate):
"Help & Contact" in Settings | Overview
---|---
Bildschirmfoto 2024-07-18 um 17 18 37 | Bildschirmfoto 2024-07-18 um 17 21 52

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: ownCloud/Client/HelpAndSupportViewController.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		])
+
+		if (VendorServices.shared.feedbackMail != nil) || (Branding.shared.feedbackURL != nil) {
+			elements.append(contentsOf: [
+				.title("Send feedback".localized),
+
+				.text("If you don't need a response and just want to send us feedback, you can also send us a mail.".localized, style: .informal, cssSelectors: [.message]),
+				.spacing(5),
+
+				.button("Send feedback".localized, action: UIAction(handler: { _ in
```
Comment (hosy) [inline ownCloud/Client/HelpAndSupportViewController.swift:96]: Would make sense to use `[weak self]`

**Review by hosy — CHANGES_REQUESTED** [review]

File: ownCloud/Client/HelpAndSupportViewController.swift (hunk 1 above)
Comment (felix-schwarz) [reply ownCloud/Client/HelpAndSupportViewController.swift]: Fixed. Thanks!

**Review by hosy — APPROVED** [review]

**Comment by jesmrec** [comment]
> for non-branded apps Settings make this view available as "Help & Contact", replacing "Documentation", "Help" and "Feedback"

Which is the expected behaviour for branded apps?  customers who use to have their own email address for managing their own feedback, might be interested in keeping that flow.

I did the following test:  i added to `Branding.plist`: 

```
<dict>
        ...
	<key>branding.send-feedback-address</key>
	<string>test@test.com</string>
        ...
</dict>
```

the `Help & Contact` from vanilla app splits in 3 sections up (the ones existing before):

Screenshot 2024-07-19 at 14 28 03

`Documentation` and `Help` point to (branded) URLs, but `Send Feedback` prompts to the survey. Shouldn't it keep the branded email address?

Don't forget that the survey names the `ownCloud iOS app`, that is static, non-brandable and unreachable for us. Customers shouldn't read that.

As reference, that was the implemented behaviour in the Android app, if helps:

https://github.com/owncloud/enterprise/issues/6594#issuecomment-2196818498

**Comment by felix-schwarz** [comment]
@jesmrec Thanks for catching this. That was a bug:
- previously there was a default value for the feedback email, but not for the feedback URL
- now it's vice versa
- therefore, when the feedback URL was set, that was preferred
- now with the default value, the URL was always used unless explicitly setting it to an empty value

The solution then was to prefer the email address now - now that it no longer has a default value. That issue is fixed with the latest commit.

**Comment by jesmrec** [comment]
> The solution then was to prefer the email address now - now that it no longer has a default value. That issue is fixed with the latest commit.

that's perfect and working fine

- [X] No branding -> survey
- [X] Branding and mail -> mail
- [X] Branding and no mail -> survey

**Comment by jesmrec** [comment]
Approved on my side

**Review by jesmrec — APPROVED** [review]

**Pull request merged**